### ISSUE TITLE
Consider Using Destructuring With Default Assignment for Default Prop Values

### ISSUE THREAD

**Issue by jsievenpiper** [Workday/canvas-kit#486]
## 💬 Questions and Help

It's not terribly uncommon in my projects to leverage Canvas Kit using aliases:

```tsx
import { Modal as CanvasModal } from '@workday/canvas-kit-react'
```

However, this breaks TypeScript's understanding of React's default props, making all props that are not explicitly marked as optional in the component's props type to become required: 

```tsx
import { Modal as CanvasModal } from '@workday/canvas-kit-react'

const Modal = <CanvasModal ... /> // ERROR: missing 'padding'
```

Per a conversation with @NicholasBoll and @anicholls, we seem to be leaning toward explicitly marking these things as optional in the props' types, and destructuring them to provide default values. This makes TypeScript very happy, and relies a bit less on weird React-isms:

```tsx
export const Modal = ({ padding = PopupPadding.m, ...props }) => //...
```

This issue is to track that discussion! 🚀

**Comment by NicholasBoll** [comment]
I agree.

The only limitation we've encountered is the Storybook Docs can't extract defaults out of props this way. Storybook uses https://github.com/styleguidist/react-docgen-typescript for extracting prop types out of Typescript code. We modified a bit here: https://github.com/Workday/canvas-kit/blob/master/.storybook/docgen-plugin.js

That would be the place to start to get defaults correctly extracted. It is very possible this is a limitation of `react-docgen-typescript`.

Is there anything stopping us from using this style in the meantime? We would loose documentation of default values in Storybook, but would fix this issue.
